homeworks/week3/Man_Man.java

```diff
@@ -0,0 +1,115 @@
+// Problem: Write two SinglyLinkedList operations: add and remove.
+public class Main
+{
+	public static void main(String[] args) {
+    SinglyLinkedList linkedList = new SinglyLinkedList();
+    linkedList.print(); // empty
+    linkedList.add(0, new Node(0));
+    linkedList.print(); // 0->null
+    linkedList.add(1, new Node(1));
+    linkedList.print(); // 0->1->null
+    linkedList.add(2, new Node(2));
+    linkedList.print(); // 0->1->2->null
+
+    System.out.println(linkedList.remove(new Node(2))); // true
+    linkedList.print(); // 0->1->null
+    System.out.println(linkedList.remove(new Node(1))); // true
+    linkedList.print(); // 0->null
+    System.out.println(linkedList.remove(new Node(0))); // true
+    linkedList.print(); // empty
+	}
+    
+  static class SinglyLinkedList {
+
+    private Node head = null;
+      // Time complexity is expected to be O(n) at worse and O(1) for best case
+      // Space complexity is O(1)
+    // Add to a specific position; do nothing if the index is out of range or e is null
+    public void add(int index, Node e) {
+      int a = range();
```

Comment:
It's important to keep a private variable containing length, rather than counting through the whole list every time this function is called